ecs-service_with_elb/main.tf

```diff
@@ -0,0 +1,45 @@
+resource "aws_ecs_task_definition" "definition" {
+  family                = "${var.service_name}_${terraform.workspace}"
+  container_definitions = "${var.template_data}"
+  network_mode          = "bridge"
+  task_role_arn         = "${aws_iam_role.role.arn}"
+}
+
+
+module "elb" {
```

Comment:
Why is this needed? afaik if you configure ECS with an ALB you can register tasks in it.
The service doesn't need to be tied to the ELB/ALB creation.

Reply:
I would indeed rather use ALB than ELB with ECS.

Reply:
@SamClinckspoor You mean ECS creates the ELB/ALB already automatically when load balancer is specified? 
I didn't work with ECS before, so maybe there's something I'm missing, but the load balancer info in here made me think the LB needs to be created separately: 
https://www.terraform.io/docs/providers/aws/r/ecs_service.html#load_balancer
Specially qouting this from http://docs.aws.amazon.com/AmazonECS/latest/developerguide/create-standard-load-balancer.html
```
After your load balancer is created, you can specify it in a service definition when you create a service. For more information, see Creating a Service.
```

Reply:
@minniux your loadbalancer is not created automatically but we have to do it. But I prefer using an ALB which you can use for multiple services/applications. So the loadbalancer is not really tied to one service.

Also by using ALB it makes the registration to ECS really easy. Happy to explain this more in depth how this works if you want.

Reply:
@minniux all the different options. http://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-load-balancing.html

As you can see ALB is much more flexible.